R/bootstrap-model.R

```diff
@@ -0,0 +1,681 @@
+#' Create a boostrap model object from an existing model
+#'
+#' @param .mod a `bbr` model object
+#' @param .suffix a suffix for the bootstrap run directory. Will be prefixed by
+#'  the model id of `.mod`.
+#' @inheritParams copy_model_from
+#' @param remove_cov,remove_tables Logical (T/F). Optionally remove `$COVARIANCE`
+#' and `$TABLE` records respectively, allowing for notably faster run times.
+#'
+#' @seealso setup_bootstrap_run summarize_bootstrap_run
+#' @return S3 object of class `bbi_nmboot_model`.
+#' @export
+new_bootstrap_run <- function(
+    .mod,
+    .suffix = "boot",
+    .inherit_tags = TRUE,
+    .overwrite = FALSE,
+    remove_cov = TRUE,
+    remove_tables = TRUE
+){
+
+  check_model_object(.mod, NM_MOD_CLASS)
+
+  model_dir <- get_model_working_directory(.mod)
+  boot_dir <- glue("{get_model_id(.mod)}-{.suffix}")
+
+  boot_run <- copy_model_from(
+    .parent_mod = .mod,
+    .new_model = boot_dir,
+    .add_tags = "BOOTSTRAP_SUBMISSION",
+    .inherit_tags = .inherit_tags,
+    .update_model_file = TRUE,
+    .overwrite = .overwrite
+  )
+
+  boot_run[[YAML_MOD_TYPE]] <- "nmboot"
+  boot_run <- save_model_yaml(boot_run)
+
+  # Change problem statement
+  prob <- glue("Bootstrap run of model {get_model_id(.mod)}")
+  modify_prob_statement(boot_run, prob)
+
+  # Optionally remove $TABLE and $COV statements here
+  if(isTRUE(remove_cov)) remove_records(boot_run, type = "covariance")
+  if(isTRUE(remove_tables)){
+    remove_records(boot_run, type = "table")
+  }else{
+    # Update table names if present (not used, but for consistency)
+    boot_run <- update_model_id(boot_run) %>% suppressMessages()
+  }
+
+  # Return read-in model to get the updated class as part of the model object
+  return(read_model(file.path(model_dir, boot_dir)))
+}
+
+
+#' Set up a bootstrap model run.
+#'
+#' Creates a new model object and re-sampled dataset per model run.
```

Comment:
`dataset` should be `datasets`

Reply:
I think dataset is right actually, though I am now realizing my use of "run" is a bit ambiguous in a few places. It's meant to suggest that one new model and one new dataset are created per "model run". Where a "bootstrap run" is a collection of [x] "model runs". 

Let me know how I should change it based on that^